clippy_lints/src/array_indexing.rs

```diff
@@ -1,10 +1,10 @@
 use rustc::lint::*;
 use rustc::middle::const_val::ConstVal;
-use rustc::ty::TyArray;
+use rustc::ty;
 use rustc_const_eval::EvalHint::ExprTypeChecked;
 use rustc_const_eval::eval_const_expr_partial;
 use rustc_const_math::ConstInt;
-use rustc::hir::*;
+use rustc::hir;
```

Comment:
I like non-glob imports better, too, but I can't promise that I won't accidentally add some, we should create a lint for that (#1228)